### PR TITLE
feat: APM server managed on 8+

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

* It adds support to launch APM server in a real managed mode for 8+

## Why is it important?

Maintaining compatibility with launching the APM Server managed in 7.x and 8+ is complicated. So it adds a new service ApmManaged to run the Elastic Agent with APM integration configured. This is the default configuration with 8+

test 7.x
```
./scripts/compose.py start 7.16.0
curl -L 127.0.0.1:8200
```

test 8+
```
./scripts/compose.py start 8.0.0
curl -L 127.0.0.1:8200
```

test Opbeans
```
./scripts/compose.py start 8.2.0 --snapshot --with-opbeans-go 
```

test APM Server from sources
```
./scripts/compose.py start 8.2.0 --apm-managed-build https://github.com/elastic/apm-server.git@main
```

closes https://github.com/elastic/apm-integration-testing/issues/1434 https://github.com/elastic/apm-integration-testing/issues/1440

### CODE REVIEW

**Comment by kuisathaverat** [comment]
I will split the implementation in two one for 7.x and another for 8.x

File: docker/kibana/kibana-8.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                name: apm
+            inputs:
+              - type: apm
+                keep_enabled: true
+                vars:
+                    - name: host
+                      value: 0.0.0.0:8200
+                      frozen: true
+                    - name: url
+                      value: http://fleet-server:8200
```
Comment (kuisathaverat) [inline docker/kibana/kibana-8.yml:38]:
```suggestion
                      value: http://apm-server:8200
```

File: scripts/tests/config/test_apm_managed.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      condition: service_healthy
+  environment:
+    FLEET_ELASTICSEARCH_HOST: http://elasticsearch:9200
+    FLEET_SERVER_ELASTICSEARCH_INSECURE: "1"
+    FLEET_SERVER_ENABLE: "1"
+    FLEET_SERVER_HOST: 0.0.0.0
+    FLEET_SERVER_INSECURE_HTTP: "1"
+    FLEET_SERVER_POLICY_ID: fleet-server-apm-policy
+    FLEET_SERVER_PORT: "8220"
+    FLEET_SERVER_SERVICE_TOKEN: AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL2VsYXN0aWMtcGFja2FnZS1mbGVldC1zZXJ2ZXItdG9rZW46bmgtcFhoQzRRQ2FXbms2U0JySGlWQQ
```
Comment (cachedout) [inline scripts/tests/config/test_apm_managed.yml:14]: Are these service tokens for the connection to Elasticsearch, as documented here? https://www.elastic.co/guide/en/elasticsearch/reference/current/service-tokens-command.html

Comment (kuisathaverat) [reply scripts/tests/config/test_apm_managed.yml:14]: yep, they are those tokens 
https://www.elastic.co/guide/en/fleet/current/add-a-fleet-server.html
```
Notes:

Make sure you download an x64 architecture installation package.
For the agent policy, choose one with a Fleet Server integration, or click Create policy to create one now. Alternatively you can [create a Fleet Server policy without using the UI](https://www.elastic.co/guide/en/fleet/8.1/create-a-policy-no-ui.html), then select the policy here.
If you choose Production deployment mode, learn how to generate certs in [Encrypt traffic in clusters with a self-managed Fleet Server](https://www.elastic.co/guide/en/fleet/8.1/secure-connections.html).
It’s recommended you generate a unique service token for each Fleet Server. For other ways to generate service tokens, see [elasticsearch-service-tokens](https://www.elastic.co/guide/en/elasticsearch/reference/8.1/service-tokens-command.html).
```

**Review by cachedout — APPROVED** [review]

File: docker/elastic-agent/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+ARG STACK_VERSION=8.2.0-SNAPSHOT
```
Comment (v1v) [inline docker/elastic-agent/Dockerfile:1]: I guess, this is something we need to bump automatically, right?

File: docker/elastic-agent/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  && version=$(mage version) \
+  && apmdir=apm-server-${version}-linux-x86_64 \
+  && builddir=build/distributions/${apmdir} \
+  && mkdir -p ${builddir} \
+  && cp -f LICENSE.txt NOTICE.txt README.md apm-server.yml ${builddir} \
+  && go build -o ${builddir}/apm-server ./x-pack/apm-server \
+  && cd build/distributions \
+  && tar -czf /apm-server.tgz ${apmdir}
+
+ARG STACK_VERSION=8.2.0-SNAPSHOT
```
Comment (v1v) [inline docker/elastic-agent/Dockerfile:28]: duplicated? I don't know if the multi-docker layouts allow to use the above defined argument in L1

**Review by v1v — APPROVED** [review]

File: docker/elastic-agent/Dockerfile (hunk 4 above)
Comment (kuisathaverat) [reply docker/elastic-agent/Dockerfile:28]: ARGs defined before the FROM are undefined after the FROM, so if you want to use them after the FROm you have to grab then again.

File: docker/elastic-agent/Dockerfile (hunk 3 above)
Comment (kuisathaverat) [reply docker/elastic-agent/Dockerfile:1]: we can, the default value is never used, this parameter is passed from the docker-compose

**Pull request merged**